src/app/inbound/mod.rs

```diff
@@ -159,44 +157,38 @@ where
                 })
             },
         ))
-        .buffer_pending(max_in_flight, DispatchDeadline::extract)
-        .service(dst_stack)
+        .into_inner()
         .make();
 
     // Share a single semaphore across all requests to signal when
     // the proxy is overloaded.
-    let admission_control = svc::builder()
-        .load_shed()
-        .concurrency_limit(max_in_flight)
-        .service(dst_router);
+    let admission_control = svc::stack(dst_router)
+        .push_concurrency_limit(max_in_flight)
+        .push_load_shed();
 
     // As HTTP requests are accepted, the `Source` connection
     // metadata is stored on each request's extensions.
     //
     // Furthermore, HTTP/2 requests may be downgraded to HTTP/1.1 per
     // `orig-proto` headers. This happens in the source stack so that
     // the router need not detect whether a request _will be_ downgraded.
-    let source_stack = svc::builder()
-        .layer(handle_time.layer())
-        .layer(super::errors::layer())
-        .layer(insert::layer(move || {
+    let source_stack = svc::stack(svc::shared(admission_control))
+        .push(orig_proto_downgrade::layer())
+        .push(insert::target::layer())
+        .push(strip_header::request::layer(super::L5D_REMOTE_IP))
+        .push(strip_header::request::layer(super::L5D_CLIENT_ID))
+        .push(strip_header::response::layer(super::L5D_SERVER_ID))
+        .push(insert::layer(move || {
             DispatchDeadline::after(dispatch_timeout)
         }))
-        .layer(strip_header::response::layer(super::L5D_SERVER_ID))
-        .layer(strip_header::request::layer(super::L5D_CLIENT_ID))
-        .layer(strip_header::request::layer(super::L5D_REMOTE_IP))
-        .layer(insert::target::layer())
-        .layer(orig_proto_downgrade::layer())
-        // disabled on purpose
```

Comment:
Looks like these were removed here, but kept in `outbound.rs`. Can we either keep or remove in both files?